### ISSUE TITLE
[speech-to-text] Customization ID Not Being Used When Specified

### ISSUE THREAD

**Issue by atonetti** [watson-developer-cloud/java-sdk#870]
The Customization ID is not being used when it is passed into the RecognizeOptions object.  I was running the same file into the sessionless recognize audio code provided by the API and getting the exact same response in terms of confidence and transcription, regardless of whether I specified the  custom language model or not.  The model I was using was trained with a corpus of 1.7 million utterances and just over 100 custom words.  The code below is from the API with the added line to specify the customization ID.  You can easily reproduce this by training a model, specifying the id in the code below BUT also add the customizationWeight parameter.  With the weight specified the output of the API call is `Cannot specify customization weight when not using customization.` I am using Java 8 and the latest version of the SDK (2.4.1)

### When reporting a bug, please be sure to include the following:

- Steps to reproduce
1. Train a custom model with some corpus/custom words
2. Run the code below using that customization ID (from the API Explorer):
```
SpeechToText service = new SpeechToText();
service.setUsernameAndPassword("{username"}, "{password}");

RecognizeOptions options = new RecognizeOptions.Builder()
  .contentType("audio/flac").timestamps(true)
  .wordAlternativesThreshold(0.9).customizationId("{customizationID}")
  .keywords(new String[]{"colorado", "tornado", "tornadoes"})
  .keywordsThreshold(0.5).build();

String[] files = {"audio-file1.flac", "audio-file2.flac"};
for (String file : files) {
  SpeechResults results = service.recognize(new File(file), options).execute();
  System.out.println(results);
}
```
3. Remove the customization ID parameter and you will see the output of the function is exactly the same

Alternatively, you can do this...
1. Train a custom model with some corpus/custom words
2. Run this code
```SpeechToText service = new SpeechToText();
service.setUsernameAndPassword("{username"}, "{password}");

RecognizeOptions options = new RecognizeOptions.Builder()
  .contentType("audio/flac").timestamps(true)
  .wordAlternativesThreshold(0.9).customizationId("{customizationID}").customizationWeight(0.5)
  .keywords(new String[]{"colorado", "tornado", "tornadoes"})
  .keywordsThreshold(0.5).build();

String[] files = {"audio-file1.flac", "audio-file2.flac"};
for (String file : files) {
  SpeechResults results = service.recognize(new File(file), options).execute();
  System.out.println(results);
}
```
3. You will notice the error that you cannot specify weight without specifying a customization model ID first.

**Comment by germanattanasio** [comment]
Thanks a lot for filing this issue! Would you like to write a patch for this? We'd be more than happy to walk you through the steps involved.

**Comment by lpatino10** [comment]
I took a look and was able to reproduce the issue, both through the SDK and through cURL. We think it might be a service-related issue and are looking into it further.

I'll update with more information once we get it figured out on that side.

**Comment by atonetti** [comment]
@lpatino10 Sounds great! Let me know.  @germanattanasio I would love to create the patch if I am allowed to.  My manager was really excited about me getting the chance to contribute to the SDK.  Don't know if you saw but I work for IBM Watson on the public sector implementations team so it would be a great experience to get under the hood of something that we use so frequently.

**Comment by arla916** [comment]
@atonetti The error "Cannot specify customization weight when not using customization" occurs when the custom model ID passed in doesn't exist.  Can you double check (using curl) that your custom model does indeed exist?

**Comment by atonetti** [comment]
@arla916 When I was doing it cURL was working for me, it was just the Java SDK that was not working

**Comment by arla916** [comment]
@atonetti The error only occurs if the custom model doesn't exist so maybe the Java SDK is passing the wrong customization ID or deleting the custom model in error.  I don't think this is an issue in the STT service.  @germanattanasio

**Comment by germanattanasio** [comment]
@atonetti can you provide the model id you are using? I would like @arla916 to validate that the model exists and see if he can reproduce the issue on his side.

**Comment by atonetti** [comment]
7598974b-9f89-47aa-b433-f31cda792246

**Comment by daniel-bolanos** [comment]
I just checked, the model does exist

**Comment by arla916** [comment]
@atonetti And you see the problem with this model, getting the same error as before when passing a customization_weight parameter?  @daniel-bolanos is the model in 'available' state?

**Comment by daniel-bolanos** [comment]
yes it is in `available` state

**Comment by atonetti** [comment]
@arla916 Yeah when I was using the Java SDK and trying to get the transcripts for an audio file there was no difference in the output (i.e. exact same transcription, same confidence scores, etc.) whether I was using the custom model or not.  I was using the sessionless audio transcription, I didn't try the other methods.  My suspicion was that the Java SDK wasn't passing the customization ID on to the service call.  That suspicion seemed to be confirmed when I again tried to get the sessionless transcript for an audio file BUT passing both the customization ID and the customization weight.  That call resulted in the error above, stating that I needed to pass a customization ID when specifying a customization weight.

**Comment by lpatino10** [comment]
Just tested with the newly released v5.0.1 and I got the expected results. Running with a customization ID vs. without produced different results and passing in the customization weight did not throw any errors. It seems the issue has been resolved.

I'm going to close this issue for now, but feel free to reopen it if you upgrade and the issue isn't fixed @atonetti
